addons/common/functions/fnc_setSpeed.sqf

```diff
@@ -0,0 +1,60 @@
+#include "script_component.hpp"
+/*
+ * Author: Ampersand
+ * Sets the movement speed mode of the given vehicle(s).
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT|ARRAY>
+ * 1: Speed <NUMBER> (default: 0)
+ *      Man: Speed modes
+ *      Vehicles: km/h
+ *
+ * 2.12   | forceSpeed              | limitSpeed
+ * _speed | neg      zero    pos    | neg        zero    pos     -(_man getSpeed "SLOW")
+ * -------|-------------------------|---------------------------------------------------------------------
+ * man    | reset    stop    yes    | reset      stop    yes     reverse, doesn't work if combat mode
+ * car    | reset    stop    yes    | reset      reset   yes
+ * tank   | reset    stop    yes    | reset      reset   yes
+ * heli   | no-eff   no-eff  no-eff | reverse    stop    yes
+ * plane  | reset    reset   yes    | min        min     yes
+ * vtol   | reset    reset   yes    | min        min     yes
+ * boat   | reset    stop    yes    | reset      stop    yes
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_vehicle, 1] call zen_common_fnc_setSpeed
+ *
+ * Public: No
+ */
+
+#define SPEED_RESET 0
+#define SPEED_FAST 1
+#define SPEED_NORMAL 2
+#define SPEED_SLOW 3
+#define SPEED_BACK 4
+
+params ["_vehicle", "_speed"];
+
+if (_vehicle isEqualType []) exitWith {
+    {
+        [_x, _speed] call FUNC(setSpeed);
```

Comment:
I wonder if `spawn` would speed things along ?

^ Granted, my laptop doesn't run both the game and the Server instance simultaneously especially fast, and on it I noticed that `forEach`+`spawn` takes a noticeable time to get things done compared to `call`, like `CBA_fnc_setLoadout` can take something like 400-600ms (looks like it) to get the squad equipped instead of next to instantly.

^ Might help when someone has a less good Internet connection, dunno.

Reply:
Oh interesting, so each one can run "in parallel"?
As for network, I tried to structure the CBA event and the function such that the selected array is pre-filtered and then only sent on network once, if I'm getting your meaning correctly.